### PR TITLE
Add upstream option for GitHub pull-request

### LINKED ISSUE #1096: Feature Request: Support upstream GitHub pullrequest
### Is your feature request related to a problem?

The two following use case triggered then need for working on a "fork" repository and then opening the pullrequest to the upstream repository

1. For security reason, I don't want to give write permission to updatecli. Instead I would like to allow the tool to work on a fork repository and then publish upstream

2. To update third party repositories, project such as homebrew, or jsonschemastore requires to open a pullrequest to submit the change. I would like to do it automatically using Updatecli

### Solution you'd like

I would like to add to add a new parameter for the  action resource `github/pullrequest`.
I am thinking of a boolean named `upstream` by default set to false

<details><summary>Manifest</summary>

```
actions:
    default:
        title: '[updatecli] Bump golangci-lint version to {{ source "default" }}'
        kind: github/pullrequest
        spec:
            automerge: true
            labels:
                - chore
                - dependencies
                - skip-changelog
        scmid: default
        upstream: true
scms:
    default:
        kind: github
        spec:
            branch: main
            email: me@olblak.com
            owner: updatecli
            repository: updatecli
            token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
            user: updatecli
            username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
        disabled: false

```

</details>

The logic would need to be implemented in the pullrequest codebase

1. Check if current repository is fork, probably by extending [codebase](https://github.com/updatecli/updatecli/blob/0e74b32c5a6e581272caad2c1eadbec6bcde971d/pkg/plugins/scms/github/main.go#L229)
2. If yes then retrieves the upstream repository information
3. Use that information when opening the pullrequest [codebase](https://github.com/updatecli/updatecli/blob/0e74b32c5a6e581272caad2c1eadbec6bcde971d/pkg/plugins/scms/github/pullrequest.go#L352)

 

### Alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by frelon**
Fix #1096 

This PR enables updatecli to open new GitHub pull-requests against parent repositories.

Since it is implied the user will not have write-access upstream we will skip setting labels/assignees/etc if not specified in the spec.

Example yaml:

```
actions:
  pull-request:
    title: "updatecli: bump go version"
    kind: github/pullrequest
    scmid: ci
    spec:
      parent: true

scms:
  ci:
    kind: github
    spec:
      user: "Elemental CI [bot]"
      owner: elemental-ci
      repository: elemental-cli
      branch: "main"
      ...
```

This pushes changes to `elemental-ci/elemental-cli` and opens the PR against the upstream `rancher/elemental-cli` as expected!

### CODE REVIEW

**Comment by olblak** [comment]
Nice one.

@frelon, did you try to update an existing pullrequest created by the updatecli pipeline? Because I have situation where the pullrequest can evolve and I would like to update the title, for example such on https://github.com/olblak/kubernetes-marketplace/pull/10 where the title display the version bump

**Comment by frelon** [comment]
@olblak I tried it initially, but got `permission denied` errors. Only setting the labels in the mutation if any are specified in the spec fixed it tho! Adding a commit to actually do the update!

**Comment by olblak** [comment]
> actually

Awesome, and it definitely seems cleaner

File: pkg/plugins/scms/github/pullrequest.go [hunk 1]
```diff
@@ -89,6 +90,8 @@ type ActionSpec struct {
 	MergeMethod string `yaml:",omitempty"`
 	// Specifies to use the Pull Request title as commit message when using auto merge, only works for "squash" or "rebase"
 	UseTitleForAutoMerge bool `yaml:",omitempty"`
+	// Specifies if a Pull Request should be sent to the upstream of a fork.
+	Upstream bool `yaml:",omitempty"`
```
[the remark this answers is not in the document]
Comment (frelon) [reply pkg/plugins/scms/github/pullrequest.go:94]: Yep, good point! Will update :+1:

**Pull request merged**